e2e_test/background_ddl/sim/basic.slt

```diff
@@ -0,0 +1,84 @@
+
+statement ok
+ALTER SYSTEM SET max_concurrent_creating_streaming_jobs TO 4;
+
+statement ok
+CREATE TABLE t (v1 int);
+
+# This is expected to run under sim mode, which is fast,
+# so we need a higher number of rows.
+statement ok
+INSERT INTO t select * from generate_series(1, 10000000);
+
+statement ok
+FLUSH;
+
+# FIXME: Ideally this can succeed at the same time as the subsequent statement.
+statement ok
+SET BACKGROUND_DDL=true;
+
+statement ok
+CREATE MATERIALIZED VIEW m1 as SELECT * FROM t;
+
+# FIXME: Ideally this can succeed at the same time as the subsequent statement.
+statement ok
+SET BACKGROUND_DDL=true;
```

Comment:
Do we have to set it every time before create any streaming job?

Reply:
Yes

Reply:
Oh sorry, I misunderstood it. It only has to be set per session normally. But under recovery, I think the session could be killed after the variable is set. In that case, the variable will revert on subsequent MVs, so I set it right before each MV. Ideally I could ensure the session + ddl is set at the same time, but I guess it can't.

Reply:
We already have some hacks for this! 😄 Does it work well? 

https://github.com/risingwavelabs/risingwave/blob/0ca0ee46a0b4efa57314dba3c355f772e9c0df2c/src/tests/simulation/src/client.rs#L136-L139

Reply:
I see, thank you I missed that.